R/epiparam_utils.R

```diff
@@ -0,0 +1,447 @@
+#' Converts rows of an epiparam object into an epidist object
+#'
+#' @description Can convert the entries (rows) of an epiparam object to one or
+#' list of several epidist objects. This allows subsetting of epiparam records
+#' into a class that can then be used to to calculate the cdf, random number
+#' generation and other methods of the epidist class
+#' (see `?epidist_distribution_functions`).
+#'
+#' @param x A epiparam object
+#'
+#' @return An epidist object or a list of epidist objects
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#'   eparam <- epiparam()
+#'   as_epidist(eparam[1, ])
+#' }
+as_epidist <- function(x) {
+
+  # check input
+  validate_epiparam(x)
+
+  # convert each epiparam row into an epidist object
+  out <- apply(x, 1, function(y) {
+    make_epidist(x = y)
+  })
+
+  # remove names of list
+  names(out) <- NULL
+
+  # unpack from list if only one epidist
+  if (length(out) == 1) out <- out[[1]]
+
+  # return list of epidist objects
+  out
+}
```

Comment:
See comment on speed and informative message on `epidist_db()`

Reply:
Will not address in this PR, but once issue #89 is resolved it should be resolved across the entire package. Please leave a comment on this issue if you have input.

Reply:
Totally fair, resolving this for now. Had not seen #89 before.